tests/more_string_manipulation/test_challenge_082.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+from exercises.more_string_manipulation import challenge_082
+
+testdata = [
+    ("All the world's a stage", 3, 10, " the wo"),
+    ("This is a poem", 1, 2, "h"),
+    ("This is another poem", 2, 2, ""),
+]
+
+testdata_fails = [
+    ("All the world's a stage", "hey", "j", TypeError),
+    ("", "hey", 2, TypeError),
+    ("Poem", 19, 23, IndexError),
+    ("Another poem", -2, -67, TypeError)
+]
+
+
+@pytest.mark.parametrize("poem, start, end, expected", testdata)
+def test_trim_characters(poem, start, end, expected):
+ 
+    trimmed_poem = challenge_082.trim_characters(poem, start, end)
+
+    assert trimmed_poem == expected
+
+@pytest.mark.parametrize("poem, start, end, expected", testdata_fails)
```

Comment:
I would separate the different kind of failure cases into different test cases. Mixing together different failure cases makes it difficult to expand test suites later or expand on different kind of failure cases.

This is also where doing TDD versus test after makes creating appropriate tests easier

Reply:
Separated into TypeError and IndexError test cases